### PR TITLE
[RISCV] Add IsSignExtendingOpW to amocas.w.

### PR DESCRIPTION
**Pull request by topperc**
New test and command line update in first commit. Changes are in the second commit. I'll commit the two commits separately.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/atomic-signext.ll [hunk 1]
```diff
@@ -2,11 +2,15 @@
 ; RUN: llc -mtriple=riscv32 -verify-machineinstrs < %s \
 ; RUN:   | FileCheck -check-prefix=RV32I %s
 ; RUN: llc -mtriple=riscv32 -mattr=+a -verify-machineinstrs < %s \
-; RUN:   | FileCheck -check-prefix=RV32IA %s
+; RUN:   | FileCheck -check-prefixes=RV32IA,RV32IA-NOZACAS %s
+; RUN: llc -mtriple=riscv32 -mattr=+a,+experimental-zacas -verify-machineinstrs < %s \
+; RUN:   | FileCheck -check-prefixes=RV32IA,RV32IA-ZACAS %s
```
Comment (jrtc27) [inline llvm/test/CodeGen/RISCV/atomic-signext.ll:7]: Why not just RV32IA and RV32IAZACAS? We don't say RV32I-NOA

**Comment by jrtc27** [comment]
How come cmpxchg_i32_monotonic_monotonic_val0 already knows it's sign-extending?

File: llvm/test/CodeGen/RISCV/atomic-signext.ll (hunk 1 above)
Comment (topperc) [reply llvm/test/CodeGen/RISCV/atomic-signext.ll:7]: Because RV32IA is used by the +a and the +a,+experimental-zacas run lines as a common prefix.

**Comment by topperc** [comment]
> How come cmpxchg_i32_monotonic_monotonic_val0 already knows it's sign-extending?

SelectionDAG knows atomic ops with i32 memory VT produce sign extended values.

File: llvm/test/CodeGen/RISCV/atomic-signext.ll (hunk 1 above)
Comment (jrtc27) [reply llvm/test/CodeGen/RISCV/atomic-signext.ll:7]: Right yes of course

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Comment by topperc** [comment]
Committed as 5446902cf2cda4fb17078d04207d2544518a49be

**Pull request closed without merging**